src/main/java/com/internet/shop/controllers/cart/CompleteOrderController.java

```diff
@@ -0,0 +1,35 @@
+package com.internet.shop.controllers.cart;
+
+import com.internet.shop.lib.Injector;
+import com.internet.shop.model.Order;
+import com.internet.shop.model.Product;
+import com.internet.shop.service.OrderService;
+import com.internet.shop.service.ShoppingCartService;
+import com.internet.shop.service.UserService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class CompleteOrderController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+    private static final Injector INJECTOR = Injector.getInstance("com.internet.shop");
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) INJECTOR.getInstance(ShoppingCartService.class);
+    private final OrderService orderService =
+            (OrderService) INJECTOR.getInstance(OrderService.class);
+    private final UserService userService =
+            (UserService) INJECTOR.getInstance(UserService.class);
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        List<Product> products = shoppingCartService.getByUserId(USER_ID).getProducts();
+        Order order = orderService.completeOrder(products, userService.get(USER_ID));
+        req.setAttribute("order", order);
```

Comment:
we should not show Order.id to our user, it's just id that will be related to database, not some kind of order number

Reply:
removed showing order id while displaying a particular order)

should order id field remain present when displaying all user's orders? otherwise the user wouldn't be able to distinguish different orders in the list

same question goes for product ids when displaying a particular order - is it OK for them to be shown?

Reply:
after discussion with Bohdan, reconsidered it. You can show OrderId in all place)sorry for that)

Reply:
no problem)